sdks/java/io/kafka/src/main/java/org/apache/beam/sdk/io/kafka/KafkaIO.java

```diff
@@ -359,20 +386,129 @@
      * of how the partitions are distributed among the splits.
      */
     public Read<K, V> withTopicPartitions(List<TopicPartition> topicPartitions) {
-      checkState(getTopics().isEmpty(), "Only topics or topicPartitions can be set, not both");
-      return toBuilder().setTopicPartitions(ImmutableList.copyOf(topicPartitions)).build();
+      return withTopicPartitions(
+          ValueProvider.StaticValueProvider.of(Joiner.on(',').join(topicPartitions)));
+    }
+
+    /**
+     * Sets a list of partitions to read from. Partitions are provided as a comma separated list of
+     * Strings in the format: topic1-partition1,topic1-partition2...
+     * This allows reading only a subset of partitions for one or more topics when (if ever) needed.
+     *
+     * <p>See {@link KafkaUnboundedSource#split(int, PipelineOptions)} for description
+     * of how the partitions are distributed among the splits.
+     */
+    public Read<K, V> withTopicPartitions(String topicPartitions) {
```

Comment:
I don't think KafkaIO should to handle parsing of string representation of `TopicPartition` objects. E.g. what if topic names have '-'? If we want the user to provide topic-partition pairs in options, probably better to add a `KafkaIOOptions` class. We could support json respresentation to specify topic-partition pairs, for e.g.

@jkff, could you take a look at the style here? e.g. in addition to `withTopoicPartitions(List<TopicPartition> partitions)`, the it adds:
 - `withTopicPartitions(String partitions)` 
 - `withTopicPartitions(ValueProvider<String>)`

Same for a few others.

Reply:
Similar comment here.